TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -73,5 +73,74 @@ class MetAPI{
                 }
             }
     }
-
+    
+    func search(parameters: [SearchParameter], completion: @escaping (SearchResponse?) -> Void) {
+        var urlString: String = self.urlBaseString
+        let urlStringSuffix: String = "public/collection/v1/search"
+        urlString.append(urlStringSuffix)
+        var searchParameters: [String : String] = self.fillingParameters(parameters: parameters)
+        self.networkManager.get(
+            urlString: urlString,
+            parameters: searchParameters) { data in
+                let jsonDecoder = JSONDecoder()
+                jsonDecoder.keyDecodingStrategy = .useDefaultKeys
+                if let data = data {
+                    do {
+                        let result = try jsonDecoder.decode(SearchResponse.self, from: data)
+                        completion(result)
+                    } catch {
+                        completion(nil)
+                    }
+                } else {
+                    completion(nil)
+                }
+            }
+    }
+    
+    private func fillingParameters (parameters: [SearchParameter]) -> [String : String] {
+        var searchParameters: [String : String] = [ : ]
+        for parameter in parameters {
+            switch parameter {
+            case .q(let text):
+                searchParameters["q"] = text
+            case .isHighlight(let bool):
+                searchParameters["isHighlight"] = self.convertBoolToString(value: bool)
+            case .title(let bool):
+                searchParameters["title"] = self.convertBoolToString(value: bool)
+            case .tags(let bool):
+                searchParameters["tags"] = self.convertBoolToString(value: bool)
+            case .departmentId(let departmentId):
+                searchParameters["departmentId"] = String(departmentId)
+            case .isOnView(let bool):
+                searchParameters["isOnView"] = self.convertBoolToString(value: bool)
+            case .artistOrCulture(let bool):
+                searchParameters["artistOrCulture"] = self.convertBoolToString(value: bool)
+            case .medium(let medium):
+                searchParameters["medium"] = medium
+            case .hasImages(let bool):
+                searchParameters["hasImages"] = self.convertBoolToString(value: bool)
+            case .geoLocation(let geoLocation):
+                searchParameters["geoLocation"] = geoLocation
+            case .dates(let dates):
+                searchParameters["dates"] = self.covertDatasToString(from: dates.from, to: dates.to)
+            }
+        }
+        return searchParameters
+    }
+    
+    private func convertBoolToString(value: Bool) -> String {
+        if value {
+            return ("true")
+        } else {
+            return ("false")
+        }
+    }
+    
+    private func covertDatasToString(from: Int, to: Int) -> String {
+        var result: String
+        result = String(describing: from)
```

Comment:
`String(describing:)` is too complicated way. Usually it is used on objects/structs. 
`String(12)` works much faster than `String(describing: 12)`

Reply:
fixed